civis/imports/socrata_helpers.py

```diff
@@ -167,4 +242,153 @@ def _read_paginated(
+    writes to pd.df then to csv. All csvs then appended into one large csv.
+    only one chunk in memory at time, as csv is on disk.
+    Inputs
+    _____________
+    dataset_id: str
+        Socrata dataset identifier
+    page_limit: int
+        number of records that can be pulled in chunk
+    date_column: 'str'
+        specifies the date column that we want to split on
+    splits: int
+        number of even splits that are made on date range.
+    Outputs
+    _____________
+    path of appened csv
+    """
+    print(dataset_id)
+    offset = 0
+    paths = []
+    min_date, max_date, date_splits = get_dates(
+        client, dataset_id, date_column="createddate", splits=splits
+    )
+
+    ### here we bring in the data, use pandas to put into dataframe and then
+    ### strip header when writing to CSV
+    ### We also build out paths for the csvs to be stored in and store the path names
+    for i in np.arange(len(date_splits)):
+        if i == len(date_splits) - 1:
+            break
+
+        where = (
+            date_column
+            + " <= '"
+            + date_splits[i]
+            + "' AND "
+            + date_column
+            + " > '"
+            + date_splits[i + 1]
+            + "'"
+        )
+
+        df = pd.DataFrame()
+
+        results = socrata_client.get(
+            dataset_id,
+            limit=page_limit,
+            content_type="csv",
+            exclude_system_fields=False,
+            offset=offset,
+            where=where
+        )
+
+        df = pd.concat([df, pd.DataFrame(results[1:], columns=results[0])])
+        print(where)
+        print(len(df))
+
+        path = "df" + str(i) + ".csv"
+        paths = np.append(path, paths)
+
+        df.to_csv(path, header=False, index=False)
+
+        df.columns = map(
+            str.lower, df.columns
+        )  ##converting headers to lower and pulling
+        headers = df.columns.str.cat(
+            sep=","
+        )  ##out headers as string to use when building big csv
+
+    csv_out = write_csv(paths, headers)
+
+    return csv_out
+
+
+def get_dates(
+    client, dataset_id: str, date_column: str = "createddate", splits: int = 70
+):
+    """
+    Uses socrata get calls to query for most recent and oldest dates
+    within the dataset, outputs those values, and creates a date range
+    based on number of splits. Can be used for date parsing.
+    Inputs
+    _____________
+    dataset_id: str
+        Socrata dataset identifier
+    splits: int
+        number of even splits that are made on date range.
+    date_column: 'str'
+        specifies the date column that we want to split on
+    Outputs
+    _____________
+    min_date: str
+        Oldest date in dataset
+    max_date: str
+        most recent date in dataset
+    date_splits: array
+        array of n dates set by splits param.
+    """
+    min_date_query = (
+        "SELECT " + date_column + " Order by " + date_column + " ASC limit 1"
```

Comment:
Seems correct, but is there a reason why `max(date_column)` doesn't work here and below?

Reply:
SoQl ( the socrata query engine) doesn't support aggregate functions. We ended up moving away from this method to a more efficient one using page_limits and offset which is not data set dependent. I kept this in for reference but can delete.

Reply:
If it's not used, I'd just as soon delete it. We can always go into git history if we need it.